src/main/java/com/conveyal/taui/models/AnalysisRequest.java

```diff
@@ -83,14 +86,15 @@ public AnalysisTask populateTask (AnalysisTask task, Project project) {
             modifications = modificationsForProject(project.accessGroup, projectId, variantIndex);
         }
 
-        // No idea how long this operation takes or if it is actually necessary
+        // Generate a checksum of all the modifications
         CRC32 crc = new CRC32();
-        crc.update(modifications.stream().map(Modification::toString).collect(Collectors.joining("-")).getBytes());
+        modifications.stream().map(JsonUtilities::objectToJsonBytes).forEach(crc::update);
         crc.update(JsonUtilities.objectToJsonBytes(this));
+        long crcValue = crc.getValue();
 
         task.scenario = new Scenario();
         // TODO figure out why we use both
-        task.jobId = String.format("%s-%s-%s", projectId, variantIndex, crc.getValue());
+        task.jobId = String.format("%s-%s-%s", projectId, variantIndex, crcValue);
```

Comment:
If we don't have variants any more, this variable name should be changed.

Reply:
I put off renaming `variant` to `scenario` since we already had a "Scenario" object in R5 that was needed here and it was messing with me a bit. Actually was the impetus for my initial foray into changing this code.

For tasks here we really don't need the Scenario object at all, we could just pass the Job ID and the modifications.

Reply:
There are also other changes I would like to make in that migration, like pulling out the "variant" arrays from the Projects & Modifications collections and creating a "Scenario" collection that it a list of modifications it contains.